src/Selenium2Library/__init__.py

```diff
@@ -184,10 +183,42 @@ def __init__(self,
         | Library `|` Selenium2Library `|` implicit_wait=5 `|` run_on_failure=Log Source | # Sets default implicit_wait to 5 seconds and runs `Log Source` on failure |
         | Library `|` Selenium2Library `|` timeout=10      `|` run_on_failure=Nothing    | # Sets default timeout to 10 seconds and does nothing on failure           |
         """
-        for base in Selenium2Library.__bases__:
-            base.__init__(self)
+        self.logger = logger
+        libraries = [
+            AlertKeywords(self),
+            BrowserManagementKeywords(self),
+            RunOnFailureKeywords(self),
+            ElementKeywords(self),
+            TableElementKeywords(self),
+            FormElementKeywords(self),
+            SelectElementKeywords(self),
+            JavaScriptKeywords(self),
+            CookieKeywords(self),
+            ScreenshotKeywords(self),
+            WaitingKeywords(self)
+        ]
+
+        DynamicCore.__init__(self, libraries)
+        self.cache = BrowserCache()
+
         self.screenshot_root_directory = screenshot_root_directory
-        self.set_selenium_timeout(timeout)
-        self.set_selenium_implicit_wait(implicit_wait)
-        self.register_keyword_to_run_on_failure(run_on_failure)
+        # self.set_selenium_timeout(timeout)
+        # self.set_selenium_implicit_wait(implicit_wait)
+        # self.register_keyword_to_run_on_failure(run_on_failure)
         self.ROBOT_LIBRARY_LISTENER = LibraryListener()
+
+    def run_keyword(self, name, args, kwargs):
+        try:
+            return DynamicCore.run_keyword(self, name, args, kwargs)
+        except Exception as e:
+            print 'run failure here'
+            raise e
+
+    def current_browser(self):
```

Comment:
This should be a property and possibly named simply `browser`. Makes usage from library components nicer when you can say `context.browser.something()` and not `context.current_browser().something()`. If the old architecture has this method, we should leave it for backwards compatibility and consider [deprecating it](https://docs.python.org/2/library/warnings.html#warnings.warn).